File: pair_sum/pair_sum.py
def pair_sum(numbers, target_sum):
  
  nums = {}
  
  for i in range(len(numbers)):
    n = numbers[i]
    
    remainder = target_sum - n
​
    if (remainder in nums):
      return (i, nums[remainder])
    
    nums[n] = i

File: pair_sum/pair_sum.js
const pairSum = (numbers, targetSum) => {
  
  
  for (let i = 0; i < numbers.length; i++) {
    for (let j = i + 1; j < numbers.length; j++) {
      const num1 = numbers[i];
      const num2 = numbers[j];
      
      if (num1 + num2 === targetSum) {
        return [i, j];
      }
    }  
  }
};
​
module.exports = {
  pairSum,
};
​